### PR TITLE
Add readiness and liveness probes to Connect API

### PR DESCRIPTION
**Pull request by mcmarkj**
I've added some readiness and liveness probes to the connect API pods. 

The operator doesn't appear to have a /health or /heartbeat endpoint so I haven't added any probes to it.

### CODE REVIEW

**Review by franklouwers — APPROVED** [review]

File: charts/connect/templates/connect-deployment.yaml [hunk 1]
```diff
@@ -78,6 +94,22 @@ spec:
...
+          readinessProbe:
+            httpGet:
+              path: /health
+              port: 8080
+            initialDelaySeconds: 15
+          {{ end }}
+          {{- if .Values.connect.probes.liveness }}
+          livenessProbe:
+            httpGet:
+              path: /health
```
Comment (thatguygriff) [inline charts/connect/templates/connect-deployment.yaml:107]: The api and sync containers have a `/heartbeat` endpoint that should be used for the `livenessProbe`

Comment (mcmarkj) [reply charts/connect/templates/connect-deployment.yaml]: Sure thing. Changed and pushed.

**Comment by jpcoenen** [comment]
My apologies for the late reply. I made some final changes to support custom port specifications and the `tls.enabled` option. Because I cannot commit to a forked branch, I opened a separate PR based off this one.

**Comment by mcmarkj** [comment]
I'll close this PR and we can work off yours.

**Pull request closed without merging**